modules/canary/policy-developer.tf

```diff
@@ -0,0 +1,62 @@
+###############################################################################
+# Policy: Developer
+# -----------------
+# A developer role can:
+# - Deploy canary functions that use CodeDeploy.
+###############################################################################
+resource "aws_iam_policy" "developer" {
+  name   = "${local.tf_group_developer_name}-canary"
+  path   = "/"
+  policy = "${data.aws_iam_policy_document.developer.json}"
+}
+
+locals {
+  codedeploy_application_name = "sls-${local.service_name}-${local.iam_stage}-Sls${replace(local.service_name, "-", "")}${local.iam_stage}DeploymentApplication"
```

Comment:
Can you add a code comment with a link to how the naming / transformation has to work?

Reply:
Totally! I need to consolidate those comments about the autogenerated name anyway 😄

Reply:
Dug a little further and added links to the exact transformation (the plugin calls a builtin Serverless function that removes non-alphanum characters). I changed the `replace` to use an equivalent regex to the one Serverless uses.